### PR TITLE
Local contexts

### PR DESCRIPTION
**Pull request by jrhoads**
## Purpose
Render local contexts as icons in commons when present in the rights metadata

closes: datacite/product-backlog/issues/4

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [x] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

**Comment by codycooperross** [comment]
@mariagould Example DOI with Local Contexts Notices to review: https://akita-4jm8co6uq-datacite.vercel.app/doi.org/10.82597/xaxa-sk06

**Review by codycooperross — COMMENTED** [review]
@jrhoads It looks like if there are only Local Contexts rights, the rights may not appear. See this example: https://akita-4jm8co6uq-datacite.vercel.app/doi.org/10.82597/xaxa-sk06 Can you take a look? 

Additionally, we've been asked to remove the rendering of the icon for the `open-to-collaborate` right. If you could remove the `open-to-collaborate` image and remove the ID from `localContexts2Icon`, that would be great. Then we should be ready to present the work to Local Contexts for a final review.

**Comment by jrhoads** [comment]
> @jrhoads It looks like if there are only Local Contexts rights, the rights may not appear. See this example: https://akita-4jm8co6uq-datacite.vercel.app/doi.org/10.82597/xaxa-sk06 Can you take a look?

Got it.  There was a clause if there wasn't a CC or an otherRights then skip rendering the licenses.  I'll extended that to also test for localContexts.  PR update incoming.

> 
> Additionally, we've been asked to remove the rendering of the icon for the `open-to-collaborate` right. If you could remove the `open-to-collaborate` image and remove the ID from `localContexts2Icon`, that would be great. Then we should be ready to present the work to Local Contexts for a final review.

Should anything be rendered? Or if it is included should it be rendered like otherRights?

**Comment by jrhoads** [comment]
@codycooperross Should be up to date here
https://commons-pr-389.datacite.vercel.app/doi.org/10.82597/xaxa-sk06

File: src/components/License/local-contexts/ci-open-to-collaborate.png
Comment (codycooperross) [inline src/components/License/local-contexts/ci-open-to-collaborate.png:1]: If could remove this image.

File: src/components/License/License.tsx [hunk 1]
```diff
@@ -18,71 +22,78 @@ type Props = {
...
+
+function isLocalContexts( rights: Rights ) {
+  return rights.rightsIdentifier && rights.rightsIdentifierScheme == "Local Contexts" && localContextsKeys.includes(rights.rightsIdentifier)
+}
+
 export const License: React.FunctionComponent<Props> = ({ rights = [] }) => {
 
   const uniqueRights = uniqBy([...rights], 'rightsIdentifier')
+
+  const localContextRigths = uniqueRights.reduce((sum, r) => {
```
Comment (codycooperross) [inline src/components/License/License.tsx:54]: Not significant, but there's a typo here in `Rigths` if you want to rename this symbol.

**Review by codycooperross — COMMENTED** [review]
Looks good! Left some notes, but we can share that preview.

**Comment by codycooperross** [comment]
> > Additionally, we've been asked to remove the rendering of the icon for the `open-to-collaborate` right. If you could remove the `open-to-collaborate` image and remove the ID from `localContexts2Icon`, that would be great. Then we should be ready to present the work to Local Contexts for a final review.
> 
> Should anything be rendered? Or if it is included should it be rendered like otherRights?

No special logic. It should appear as any right would if not detected as a Local Contexts or CC right; i.e. as a badge.

File: src/components/License/License.tsx (hunk 1 above)
Comment (jrhoads) [reply src/components/License/License.tsx]: Good catch.  Done.

File: src/components/License/local-contexts/ci-open-to-collaborate.png
Comment (jrhoads) [reply src/components/License/local-contexts/ci-open-to-collaborate.png:1]: Done

**Comment by jrhoads** [comment]
> > > Additionally, we've been asked to remove the rendering of the icon for the `open-to-collaborate` right. If you could remove the `open-to-collaborate` image and remove the ID from `localContexts2Icon`, that would be great. Then we should be ready to present the work to Local Contexts for a final review.
> > 
> > 
> > Should anything be rendered? Or if it is included should it be rendered like otherRights?
> 
> No special logic. It should appear as any right would if not detected as a Local Contexts or CC right; i.e. as a badge.

That is the behavior.  Done

**Pull request merged**